platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java

```diff
@@ -777,18 +777,21 @@ private interface ClassValidator {
     }
 
     private static class ClassGenerationHandler {
+        @SuppressWarnings("UnusedVariable") // used in subclasses
```

Comment:
💅  Can we suppress for the variable instead of the whole method here and below?

💭 I saw something similar in other places and I wonder if it strips almost all usefulness from the check.